sdks/python/apache_beam/io/avroio.py

```diff
@@ -94,6 +95,7 @@ def __init__(self, file_pattern, min_bundle_size=0):
     self._avro_schema = None
     self._codec = None
     self._sync_marker = None
+    self._progress_lock = threading.Lock()
```

Comment:
does Python have any annotations you would use like Java's `@GuardedBy` to enable tools to identify locking bugs?